### PR TITLE
Add coverage for content view version delete functionality for API

### LINKED ISSUE #2471: Automate content view version delete in api
please automate:

DELETE /katello/api/content_view_versions/:id
Remove content view version
Params
Param name 	Description
id
required 	

Content view version identifier

Value: string from 2 to 128 characters containting only alphanumeric characters, space, '_', '-' with no leading or trailing space..

### PR DESCRIPTION
**Pull request by oshtaier**
Closes #2471

```
nosetests tests/foreman/api/test_contentviewversion.py -m test_delete
...
----------------------------------------------------------------------
Ran 3 tests in 160.507s

OK
```

Not sure whether that code can be merged right now as I am using latest Nailgun version from repository

### CODE REVIEW

File: tests/foreman/api/test_contentviewversion.py [hunk 1]
```diff
@@ -66,16 +70,65 @@ def test_delete_version(self):
...
+        self.assertEqual(len(content_view.version), 1)
+        cvv = content_view.version[0].read()
+        self.assertEqual(len(cvv.environment), 1)
+        lce = entities.LifecycleEnvironment(organization=org).create()
+        content_view.version[0].promote(lce.id)
+        cvv = content_view.version[0].read()
+        self.assertEqual(len(cvv.environment), 2)
+        # Delete the content-view version from selected environments
+        for i in [1, 0]:
+            content_view.delete_from_environment(cvv.environment[i].id)
```
Comment (abalakh) [inline tests/foreman/api/test_contentviewversion.py]: how about
```python
for env in reversed(cvv.environment):
    content_view.delete_from_environment(env.id)
```
?

Comment (oshtaier) [inline tests/foreman/api/test_contentviewversion.py]: I thought in that way before, but that approach seems less readable. Anyway, as usually, if someone caught same idea, so it is necessary to introduce that fix. Will do soon...

File: tests/foreman/api/test_contentviewversion.py [hunk 2]
```diff
@@ -66,16 +70,65 @@ def test_delete_version(self):
...
+        org = entities.Organization().create()
+        content_view = entities.ContentView(organization=org).create()
+        # Publish content view
+        content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 1)
+        cvv = content_view.version[0].read()
+        self.assertEqual(len(cvv.environment), 1)
+        lce = entities.LifecycleEnvironment(organization=org).create()
+        content_view.version[0].promote(lce.id)
```
Comment (abalakh) [inline tests/foreman/api/test_contentviewversion.py]: ```cvv``` and ```lce``` are used only once. You can drop them to smth like:
```python
        self.assertEqual(len(content_view.version[0].read().environment), 1)
        content_view.version[0].promote(
            entities.LifecycleEnvironment(organization=org).create().id
        )
```
Not sure if that looks better though.

File: tests/foreman/api/test_contentviewversion.py [hunk 3]
```diff
@@ -66,16 +70,65 @@ def test_delete_version(self):
...
+        cvv = content_view.version[0].read()
+        self.assertEqual(len(cvv.environment), 1)
         # Delete the content-view version from selected env
-        content_view.delete_from_environment(lc_env_id)
+        content_view.delete_from_environment(cvv.environment[0].id)
         # Delete the version
         content_view.version[0].delete()
+        # Make sure that content view version is really removed
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 0)
```
Comment (abalakh) [inline tests/foreman/api/test_contentviewversion.py]: ```content_view``` variable won't be used anywhere else. You can drop it with:
```python
    self.assertEqual(len(content_view.read().version), 0)
```
Same for the rest tests.

**Comment by abalakh** [comment]
ACK, pending comments.

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/api/test_contentviewversion.py -m test_delete
...
----------------------------------------------------------------------
Ran 3 tests in 172.022s

OK
```

File: tests/foreman/api/test_contentviewversion.py (hunk 1 above)
Comment (Ichimonji10) [inline tests/foreman/api/test_contentviewversion.py]: This is neat. I've never used `reversed` before.

**Comment by Ichimonji10** [comment]
ACK

**Pull request merged**